File: app/src/main/java/com/damianogiusti/cleanandroid/ui/home/MainView.java
package com.damianogiusti.cleanandroid.ui.home;

import com.molo17.cloudnotes.mvp.View;
import com.molo17.cloudnotes.viewmodel.ProvinceViewModel;

import java.util.List;

/**
 * Created by Damiano Giusti on 03/05/17.
 */
public interface MainView extends View {

    void showMessage(String message);

    void showProvinces(List<ProvinceViewModel> provincesList);

    void showProgress();

    void hideProgress();

    void showError(String errorMessage);
}

File: app/src/main/java/com/damianogiusti/cleanandroid/ui/home/MainView.kt
package com.damianogiusti.cleanandroid.ui.home


import com.damianogiusti.cleanandroid.mvp.View
import com.damianogiusti.cleanandroid.viewmodel.ProvinceViewModel

/**
 * Created by Damiano Giusti on 03/05/17.
 */
interface MainView : View {

    fun showMessage(message: String)

    fun showLoading()

    fun hideLoading()

    fun showList(list: List<ProvinceViewModel>)

    fun showError(errorMessage: String)
}
